### PR TITLE
Fix InfluxDB write support to work with InfluxDB 0.9.x.

### PR DESCRIPTION
**Pull request by juliusv**
Because the InfluxDB client library currently pulls in multiple MBs of
unnecessary dependencies, I have modified and cut up the vendored
version to only pull in the few pieces that are actually needed.

On InfluxDB's side, this dependency issue is tracked in:

https://github.com/influxdb/influxdb/issues/3447

Hopefully, it will be resolved soon.

### CODE REVIEW

File: storage/remote/influxdb/client.go [hunk 1]
```diff
@@ -14,148 +14,97 @@
...
 	"github.com/prometheus/common/model"
 	"github.com/prometheus/log"
 
-	"github.com/prometheus/prometheus/util/httputil"
-)
-
-const (
-	writeEndpoint   = "/write"
-	contentTypeJSON = "application/json"
+	idb_client "github.com/influxdb/influxdb/client"
```
Comment (beorn7) [inline storage/remote/influxdb/client.go]: Why underscore? Strangely, golint doesn't complain. Perhaps I'm missing something?

Comment (juliusv) [inline storage/remote/influxdb/client.go]: Neither camelCase nor under_scores are common for package names in Go in general, but for renamed imports we already use underscores for this at least here: https://github.com/prometheus/prometheus/blob/master/storage/local/index/leveldb.go#L20-L23

File: storage/remote/influxdb/client.go [hunk 2]
```diff
@@ -14,148 +14,97 @@
...
-func tagsFromMetric(m model.Metric) model.LabelSet {
-	tags := make(model.LabelSet, len(m)-1)
+func tagsFromMetric(m model.Metric) map[string]string {
+	tags := make(map[string]string, len(m)-1)
 	for l, v := range m {
 		if l == model.MetricNameLabel {
 			continue
 		}
-		tags[l] = v
+		tags[string(l)] = string(v)
```
Comment (beorn7) [inline storage/remote/influxdb/client.go]: Obviously not touched by this PR, but why the inverted logic here?

```
if l != model.MetricNameLabel {
    tags[string(l)] = string(v)
}
```

Comment (juliusv) [inline storage/remote/influxdb/client.go]: No particular reason - changed now.

File: storage/remote/queue_manager.go [hunk 3]
```diff
@@ -92,15 +93,22 @@ func NewStorageQueueManager(tsdb StorageClient, queueCapacity int) *StorageQueue
 		sendLatency: prometheus.NewSummary(prometheus.SummaryOpts{
 			Namespace:   namespace,
 			Subsystem:   subsystem,
-			Name:        "sent_latency_milliseconds",
+			Name:        "send_latency_milliseconds",
```
Comment (beorn7) [inline storage/remote/queue_manager.go]: If we change the metric name anyway, why not switch to seconds at the same time?

File: storage/remote/queue_manager.go [hunk 4]
```diff
@@ -92,15 +93,22 @@ func NewStorageQueueManager(tsdb StorageClient, queueCapacity int) *StorageQueue
...
 			ConstLabels: constLabels,
 		}),
-		sendErrors: prometheus.NewCounter(prometheus.CounterOpts{
+		failedBatches: prometheus.NewCounter(prometheus.CounterOpts{
 			Namespace:   namespace,
 			Subsystem:   subsystem,
-			Name:        "sent_errors_total",
-			Help:        "Total number of errors sending sample batches to the remote storage.",
+			Name:        "failed_batches_total",
+			Help:        "Total number of sample batches that encountered an error while sending to the remote storage.",
```
Comment (beorn7) [inline storage/remote/queue_manager.go]: sending -> being sent ?

File: storage/remote/queue_manager.go [hunk 5]
```diff
@@ -92,15 +93,22 @@ func NewStorageQueueManager(tsdb StorageClient, queueCapacity int) *StorageQueue
...
+			ConstLabels: constLabels,
+		}),
+		failedSamples: prometheus.NewCounter(prometheus.CounterOpts{
+			Namespace:   namespace,
+			Subsystem:   subsystem,
+			Name:        "failed_samples_total",
+			Help:        "Total number of samples that encountered an error while sending to the remote storage.",
```
Comment (beorn7) [inline storage/remote/queue_manager.go]: sending -> being sent ?

File: storage/remote/remote.go [hunk 6]
```diff
@@ -14,12 +14,15 @@
...
 
 import (
+	"net/url"
 	"sync"
 	"time"
 
 	"github.com/prometheus/client_golang/prometheus"
 	"github.com/prometheus/common/model"
 
+	idb_client "github.com/influxdb/influxdb/client"
```
Comment (beorn7) [inline storage/remote/remote.go]: Same underscore question as above.

File: storage/remote/queue_manager.go (hunk 4 above)
Comment (juliusv) [inline storage/remote/queue_manager.go]: I was thinking from the perspective of the storage queue at the end of the sentence again, but I guess that makes more sense :)

File: storage/remote/queue_manager.go (hunk 3 above)
Comment (juliusv) [inline storage/remote/queue_manager.go]: Updated this and the rest again.

File: storage/remote/remote.go (hunk 6 above)
Comment (juliusv) [inline storage/remote/remote.go]: Same answer :)

File: storage/remote/influxdb/client.go (hunk 1 above)
Comment (beorn7) [inline storage/remote/influxdb/client.go]: How about `influxclient`?

The underscore looks sooo weird in Go code... (not feeling strongly, though... if golint is fine with it...)

Comment (juliusv) [inline storage/remote/influxdb/client.go]: Don't care either - updated to `influxclient`.

**Comment by beorn7** [comment]
Thanks for the heroic effort of vendoring out the relevant parts.

Another thought on `influxclient` vs `idb_client`: Perhaps just `influxdb` would be best. Looking at the exported names, that reads really nicely:
- influxdb.NewClient
- influxdb.Point
- ...

Not sure if the collisions with our own package name matter. Could also use just `influx` in that case.

I leave it to your judgement.


:+1: in any case.

**Comment by juliusv** [comment]
`influxdb` causes a conflict, so I went with `influx` now.

**Pull request merged**